interactive_templates/templates/v2/tests/test_measures.py

```diff
@@ -24,3 +27,129 @@ def test_redact_and_round_column(data_frame):
     # Rather than reimplement the rounding function, we test that values are multiples
     # of 10.
     assert rounded_data_frame.loc[:, "A"].mod(10).sum() == 0
+
+
+@pytest.fixture
+def filter_data_df():
+    return pd.DataFrame(
+        {
+            "A": [1, 2, 3, 4, 5],
+            "B": ["a", "b", "c", "d", "e"],
+            "C": [10, 20, 30, 40, 50],
+        }
+    )
+
+
+def test_filter_data(filter_data_df):
+    filters = {
+        "A": [1, 3, 5, 6],
+        "B": ["a", "d", "a"],
+        "C": [10, 20, 50, 10],
+    }
+
+    exp = pd.DataFrame(
+        {
+            "A": [1],
+            "B": ["a"],
+            "C": [10],
+        }
+    )
+
+    obs = measures.filter_data(filter_data_df, filters)
+    assert exp.equals(obs)
+
+
+@st.composite
+def input_df(draw):
+    nrows = draw(st.integers(min_value=20, max_value=20))
+
+    patient_id = column(
+        name="patient_id",
+        elements=st.integers(min_value=1, max_value=1000),
+        unique=True,
+    )
+
+    sex = column(name="sex", elements=st.sampled_from(["M", "F"]), unique=False)
+
+    event_measure = column(
+        name="event_measure",
+        elements=st.integers(min_value=0, max_value=1),
+        unique=False,
+    )
+
+    population = column(
+        name="population", elements=st.integers(min_value=1, max_value=1), unique=False
+    )
+
+    imd = column(
+        name="imd", elements=st.integers(min_value=1, max_value=5), unique=False
+    )
+
+    return draw(
+        data_frames(
+            [patient_id, event_measure, population, sex, imd],
+            index=range_indexes(min_size=nrows, max_size=nrows),
+        )
+    )
+
+
+@given(df=input_df())
+def test_calculate_total_counts(df):
+    date = "2022-01-01"
+    df["date"] = date
+
+    obs = measures.calculate_total_counts(df, date, group="total", group_value="total")
+
+    # assert that both the group and group_value are "total" for every row
```

Comment:
I don't think this comment is very useful; it says the same as the code, in English.